src/libraries/System.Reflection.Metadata/tests/Metadata/TypeSystem/TypeDefinitionTests.cs

```diff
@@ -32,5 +33,58 @@ public void ValidateTypeDefinitionIsNestedWindowsProjection()
                 Assert.Equal(typeDef.Attributes.IsNested(), typeDef.IsNested);
             }
         }
+
+        [Fact]
+        public void ValidateMemberDeclaringType()
+        {
+            var reader = MetadataReaderTests.GetMetadataReader(Misc.Members);
+
+            Assert.NotEmpty(reader.GetTypesWithEvents());
+            Assert.NotEmpty(reader.GetTypesWithProperties());
+
+            foreach (var typeDefHandle in reader.TypeDefinitions)
+            {
+                var typeDef = reader.GetTypeDefinition(typeDefHandle);
+                foreach (var fieldHandle in typeDef.GetFields())
+                {
+                    var field = reader.GetFieldDefinition(fieldHandle);
+                    Assert.Equal(typeDefHandle, field.GetDeclaringType());
+                }
+                foreach (var methodHandle in typeDef.GetMethods())
+                {
+                    var method = reader.GetMethodDefinition(methodHandle);
+                    Assert.Equal(typeDefHandle, method.GetDeclaringType());
+                }
+                foreach (var eventHandle in typeDef.GetEvents())
+                {
+                    var eventDef = reader.GetEventDefinition(eventHandle);
+                    Assert.Equal(typeDefHandle, eventDef.GetDeclaringType());
+                }
+                foreach (var propertyHandle in typeDef.GetProperties())
+                {
+                    var property = reader.GetPropertyDefinition(propertyHandle);
+                    Assert.Equal(typeDefHandle, property.GetDeclaringType());
+                }
+            }
+        }
+
+#if NET
+        [ActiveIssue("https://github.com/dotnet/runtime/pull/111642", TestRuntimes.Mono)]
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotNativeAot))]
```

Comment:
All existing tests for System.Reflection.Metadata use standalone assemblies so that they are compatible with NAOT and other form-factors that do not preserve ECMA metadata. We do not want to be introducing the test based on runtime assembly for System.Reflection.Metadata.

Is there a problem with adding some nested types to the existing Members assembly?

Reply:
https://github.com/dotnet/runtime/pull/111646#discussion_r1943954036

Reply:
> Would it be better to add some nested types to this assembly instead?

That's how I interpreted the "this assembly" part of that comment.

Reply:
I am sorry for not being clear. I meant to add it to the existing Members.dll assembly.

> https://github.com/dotnet/runtime/pull/111646#discussion_r1943954036

Members.dll is old and it was compiled with the .NET Framework compiler. You can use .NET Framework compiler to recompile the .dll to avoid unrelated diffs, e.g.:

C:\runtime\src\libraries\System.Reflection.Metadata\tests\Resources\Misc>c:\Windows\Microsoft.NET\Framework\v4.0.30319\csc /target:library Members.cs

Once you add a nested type to the .dll, there will be a few tests that need updating (e.g. the test that enumerates all members of the .dll), but I do not expect the relocbloc test to fail.

Reply:
I have reproduced it locally. It is fine to update the expected value in the failing test. I have pushed the change into this PR.

Reply:
Thank you!